### ISSUE TITLE
Cfg file crashes

### ISSUE THREAD

**Issue by ikajic** [nengo/nengo_gui#883]
I don't know what exactly causes this error, but I had it twice in the last two days. I'm working on the master. This is the output I am seeing in the browser* and in the console:
```
Traceback (most recent call last):
  File "/home/ivana/phd/workspace/nengo_gui/nengo_gui/server.py", line 235, in do_GET
    self.http_GET()
  File "/home/ivana/phd/workspace/nengo_gui/nengo_gui/server.py", line 251, in http_GET
    response = getattr(self, command)()
  File "/home/ivana/phd/workspace/nengo_gui/nengo_gui/guibackend.py", line 92, in auth_checked
    return fn(inst)
  File "/home/ivana/phd/workspace/nengo_gui/nengo_gui/guibackend.py", line 191, in serve_main
    page = self.server.create_page(filename, reset_cfg=reset_cfg)
  File "/home/ivana/phd/workspace/nengo_gui/nengo_gui/guibackend.py", line 390, in create_page
    reset_cfg=reset_cfg)
  File "/home/ivana/phd/workspace/nengo_gui/nengo_gui/page.py", line 110, in __init__
    self.load()
  File "/home/ivana/phd/workspace/nengo_gui/nengo_gui/page.py", line 172, in load
    self.config = self.load_config()
  File "/home/ivana/phd/workspace/nengo_gui/nengo_gui/page.py", line 287, in load_config
    v.attach(page=self, config=config[v], uid=k)
  File "/home/ivana/phd/workspace/nengo_gui/nengo_gui/components/spa_plot.py", line 21, in attach
    self.vocab_out.include_pairs = config.show_pairs
AttributeError: 'int' object has no attribute 'include_pairs'
```
I am having some guesses as to what happened, but I'm not sure: it may be related to creating a spa.State which doesn't have any semantic pointers (i.e. not passing any vocabulary when creating the object). This is a wild guess and if I find out anything more I'll update the issue and the title.

 For now, I'm "solving" the problem by removing the .cfg file.

\* browser is actually two different browsers, but they both show the same output:
- Chromium 55.0.2883.75 Built on 8.6, running on Debian 8.7
- Mozilla Firefox 45.6.0

**Comment by jgosmann** [comment]
Can you post your or link to you model code?

**Comment by jgosmann** [comment]
Might be related to #640.

**Comment by ikajic** [comment]
The code is [here](https://gist.github.com/ikajic/cf7b6897caaf8cbb0b6db55319e3c80e). It's part of a bigger project and to run it other files are needed, but this gives you an idea of what's happening. I will make the project publicly available next week, right now I'm still working on this for my course. You can also run some parts of that model by commenting out bits which use external data.

**Comment by jgosmann** [comment]
Try to use `nengo.spa.SPA` instead of `nengo.Network` [in line 10](https://gist.github.com/ikajic/cf7b6897caaf8cbb0b6db55319e3c80e#file-semnet-code-L10), otherwise SPA modules are not guaranteed to work properly. Also, SPA modules ([like in line 40](https://gist.github.com/ikajic/cf7b6897caaf8cbb0b6db55319e3c80e#file-semnet-code-L40)) need to me assigned to the `nengo.spa.SPA` instance (e.g. `model.sentence = spa.State(....)`). (You should get a warning or exception about the latter once you change `nengo.Network` to `nengo.spa.SPA`.)

The new-spa rewrite will improve these things a little bit by allowing SPA modules to be used in normal Nengo networks. (Waiting on a last PR review right now that is blocking my work on that ...)

**Comment by jgosmann** [comment]
Is this still happening or has this been solved?

**Comment by ikajic** [comment]
After I integrated changes you proposed it has happened once more. I still
do not know what causes this.

**Comment by jgosmann** [comment]
Can you provide updatet code?
I assume this happens when you try to open a semantic pointer plot? I would expect this to also be reproducible, i.e. always happens when opening a semantic pointer plot for the same module.

**Comment by ikajic** [comment]
The updated code is here:  https://github.com/ikajic/cs886proj-affect-neuralnet/blob/master/network.py

> I assume this happens when you try to open a semantic pointer plot? 

I am guessing that it is related to displaying a SP plot or SP cloud, but I am not sure. When I had it happening, I opened a SP plot  (maybe did some other things) and at some point noticed that couldn't run the simulation (nothing would happen when I press play) and if I look in the console there would be the error I posted. Now, when that error was printed in the console (before or after I opened a plot or when I clicked on something) I don't know.

> I would expect this to also be reproducible, i.e. always happens when opening a semantic pointer plot for the same module.

My initial guess is that it has to do with passing an empty vocabulary to a `spa.State` and displaying that state, but I tried this again and it didn't happen. So it seems that it's not just displaying a plot, but there are some other things...

**Comment by jgosmann** [comment]
> My initial guess is that it has to do with passing an empty vocabulary to a spa.State

It's more likely not setting a vocabulary (only the dimensions) on a spa.State (or other spa.Module). There is some magic happening which should replace the integer (dimensions) with the actual vocabulary, but this replacement seems to not happen.

**Comment by jgosmann** [comment]
All the vocabularies are properly set now which means that it should not be possible for the above exception to occur.
1. Are you sure that you still got the same exception? (There are lots of other bugs in the GUI ...)
2. Are you sure that you didn't have an unassigned SPA module in the code that is not any longer in the code you posted?

**Comment by ikajic** [comment]
>  Are you sure that you still got the same exception? (There are lots of other bugs in the GUI ...)

It could have been a different one, but I am pretty certain it was the same as the one I posted.

> Are you sure that you didn't have an unassigned SPA module in the code that is not any longer in the code you posted?

If by "unassigned SPA module" you mean a SPA module that was not referenced to by `model.` then yes, I am sure that there was no such module in my code.

Also, I tried recreating the scenario with a SPA module that has an empty vocabulary and displaying SP cloud/plot and couldn't get that error with that only, so it seems there is something else triggering this.

**Comment by jgosmann** [comment]
You need to not set a vocab at all to get the error:
```python
import nengo
from nengo import spa

net = nengo.Network('tmp')
with net as model:
    d=16
    s = spa.State(dimensions=d)
```

**Comment by ikajic** [comment]
Right, I can reproduce that. Then, I'd say this is very likely the source
of error I posted above

**Comment by jgosmann** [comment]
Then this is a duplicate of #640.
